test/unit/test_loss_image.py

```diff
@@ -0,0 +1,78 @@
+"""
+Tests for deepreg/model/loss/image.py in
+pytest style
+"""
+import numpy as np
+import pytest
+import tensorflow as tf
+
+import deepreg.model.loss.image as image
+
+
+def assertTensorsEqual(x, y):
+    """
+    given two tf tensors return True/False (not tf tensor)
+    tolerate small errors
+    :param x:
+    :param y:
+    :return:
+    """
+    return tf.reduce_max(tf.abs(x - y)).numpy() < 1e-6
+
+
+def test_dissimilarity_fn():
+    """
+    Testing computed dissimilarity function by comparing to precomputed, the dissimilarity function can be either
+    normalized cross correlation or sum square error function
+    """
+    tensor_true = np.array(range(12)).reshape((2, 1, 2, 3))
+    tensor_pred = 0.6 * np.ones((2, 1, 2, 3))
+    tensor_true = tf.convert_to_tensor(tensor_true)
+    tensor_pred = tf.convert_to_tensor(tensor_pred)
+    tensor_true = tf.cast(tensor_true, dtype=tf.float32)
```

Comment:
As mentioned in https://github.com/DeepRegNet/DeepReg/issues/197, I suggest to move `dtype` into the `convert_to_tensor` function.